tools/jnimarshalmethod-gen/App.cs

```diff
@@ -123,21 +123,18 @@ static Expression CreateRegistration (string method, string signature, LambdaExp
 
 		static void AddRegisterNativeMembers (TypeBuilder dt, ParameterExpression targetType, List<Expression> registrationElements)
 		{
-			var type    = Expression.Parameter (typeof (JniType),   "type");
-			var members = Expression.Parameter (typeof (string),    "members");
-
-			var methods = Expression.Variable (typeof (JniNativeMethodRegistration[]),  "methods");
+			var args    = Expression.Parameter (typeof (JniNativeMethodRegistrationArguments),   "args");
 
 			var body = Expression.Block (
-					new[]{targetType, methods},
+					new[]{targetType},
 					Expression.Assign (targetType, Expression.Call (Type_GetType, Expression.Constant (dt.FullName))),
-					Expression.Assign (methods, Expression.NewArrayInit (typeof(JniNativeMethodRegistration), registrationElements.ToArray ())),
-					Expression.Call (type, JniType_RegisterNativeMethods, methods));
+					Expression.Call (args, JniNativeMethodRegistrationArguments_AddRegistrations, Expression.NewArrayInit (typeof (JniNativeMethodRegistration), registrationElements.ToArray ())));
 
-			var lambda  = Expression.Lambda<Action<JniType, string>> (body, new[]{ type, members });
+			var lambda  = Expression.Lambda<Action<JniNativeMethodRegistrationArguments>> (body, new[]{ args });
 
-			var rb = dt.DefineMethod ("__RegisterNativeMembers",
```

Comment:
It's *probably* still a good idea to leave this as `__RegisterNativeMembers`, or use some other "unique" name, because the type generated will also contain names under *user* control,  as the marshal method names mirror the method names to register.

Otherwise we need to rely on the parameter signature being unique, which is also *probably* fine, but doesn't make me feel entirely good.